File: aat-android/src/main/java/ch/bailu/aat/map/layer/MapLayerInterface.java
package ch.bailu.aat.map.layer;

import org.mapsforge.core.model.Point;

import ch.bailu.aat.map.Attachable;
import ch.bailu.aat.map.MapContext;
import ch.bailu.aat_lib.preferences.OnPreferencesChanged;

public interface MapLayerInterface extends Attachable,
        OnPreferencesChanged {

    void onLayout(boolean changed, int l, int t, int r, int b);
    void drawInside(MapContext mcontext);
    void drawForeground(MapContext mcontext);

    boolean onTap(Point tapPos);
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/map/layer/MapLayerInterface.kt
package ch.bailu.aat_lib.map.layer

import ch.bailu.aat_lib.map.Attachable
import ch.bailu.aat_lib.map.MapContext
import ch.bailu.aat_lib.preferences.OnPreferencesChanged
import ch.bailu.aat_lib.util.Point

interface MapLayerInterface : Attachable, OnPreferencesChanged {
    fun onLayout(changed: Boolean, l: Int, t: Int, r: Int, b: Int)
    fun drawInside(mcontext: MapContext)
    fun drawForeground(mcontext: MapContext)
    fun onTap(tapPos: Point): Boolean
}
